### PR TITLE
ADEN-9634 | Add incontent_boxads to UCP

### PR DESCRIPTION
**Pull request by bckatarzyna**

### CODE REVIEW

File: src/ad-products/utils/fmr-rotator.ts [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+		refreshDelay: 10000,
+		refreshLimit: 20,
+		startPosition: 0,
+	};
+	navbarManager = new NavbarManager(document.getElementById('globalNavigation'));
+	rotatorListener: string;
+	recSelector: string;
+	currentRecNode: HTMLElement;
+
+	constructor(private slotName: string, private fmrPrefix: string, private btRec) {}
```
Comment (Bielik20) [inline src/ad-products/utils/fmr-rotator.ts]: That `btRec` needs to change

File: src/ad-products/utils/fmr-rotator.ts [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+	events,
+	eventService,
+	scrollListener,
+	slotService,
+	utils,
+} from '@ad-engine/core';
+import { universalAdPackage } from '../templates/uap';
+import { NavbarManager } from './navbar-manager';
+
+export class FmrRotator {
```
Comment (fraszczakszymon) [inline src/ad-products/utils/fmr-rotator.ts]: Can we make it more general? Like `SlotRotator`?

File: src/ad-engine/services/slot-injector.ts [hunk 3]
```diff
@@ -22,7 +22,7 @@ function findNextSuitablePlace(
 	return null;
 }
 
-function insertNewSlot(
+export function insertNewSlot(
```
[the remark this answers is not in the document]
Comment (bckatarzyna) [reply src/ad-engine/services/slot-injector.ts]: This change is from different branch (https://github.com/Wikia/ad-engine/pull/591/files#diff-39ebafe35e8eaac750e1c3a4bb5a024fR25)

File: platforms/ucp/setup/dynamic-slots/ucp-dynamic-slots.setup.ts [hunk 4]
```diff
@@ -1,18 +1,48 @@
...
 	}
 
-	private injectTopLeaderboardPlaceholder(): void {
-		const container = document.querySelector('.WikiaPage');
-		const topLeadeboardWrapper = document.createElement('div');
-		const parentElement = container.parentNode;
+	private static appendIncontentBoxad(slotConfig: SlotConfig): void {
+		// TODO change to something like:
+		// TODO const isSupportedPageType = 'article'.indexOf(context.get('wiki.targeting.pageType')) !== -1;
+		// TODO when context fixed
```
[the remark this answers is not in the document]
Comment (bckatarzyna) [reply platforms/ucp/setup/dynamic-slots/ucp-dynamic-slots.setup.ts]: Ok, I'll remove them, let's assume that ICBs will only appear when `recirculation-rail` is present

File: src/ad-products/utils/fmr-rotator.ts (hunk 1 above)
Comment (bckatarzyna) [reply src/ad-products/utils/fmr-rotator.ts]: Too big change for now

File: src/ad-products/utils/fmr-rotator.ts (hunk 2 above)
Comment (bckatarzyna) [reply src/ad-products/utils/fmr-rotator.ts]: It has some fmr-specific behaviour like e.g. startPosition. I'm not sure if we want to have it for all slots in the possible future.

File: platforms/ucp/setup/dynamic-slots/ucp-dynamic-slots.setup.ts [hunk 5]
```diff
@@ -1,18 +1,52 @@
...
+		container.id = slotName;
+		parentContainer.appendChild(container);
+		rotator.rotateSlot();
 	}
 
-	private injectTopLeaderboardPlaceholder(): void {
-		const container = document.querySelector('.WikiaPage');
-		const topLeadeboardWrapper = document.createElement('div');
-		const parentElement = container.parentNode;
+	private appendIncontentBoxad(slotConfig: SlotConfig): void {
```
[the remark this answers is not in the document]
Comment (bckatarzyna) [reply platforms/ucp/setup/dynamic-slots/ucp-dynamic-slots.setup.ts]: done

**Pull request merged**